File: doodle/src/main/java/io/github/doodle/enums/MediaType.java
package io.github.doodle.enums;

public enum MediaType {
    UNKNOWN,

    MP4,
    MOV,
    _3GP,
    MKV,

    GIF,
    JPG,
    PNG,
    PNG_NO_ALPHA,
    WEBP_LOSSY,
    WEBP_LOSSY_NO_ALPHA,
    WEBP_LOSSLESS,
    WEBP_LOSSLESS_NO_ALPHA,
    WEBP_ANIMATED,
    BMP,
    HEIF;

    public boolean isVideo() {
        return this == MP4 || this == MOV || this == _3GP || this == MKV;
    }

    public boolean isLossy() {
        return this == JPG || this == WEBP_LOSSY || this == WEBP_LOSSY_NO_ALPHA;
    }

    public boolean noAlpha() {
        return this == JPG ||
                this == PNG_NO_ALPHA ||
                this == WEBP_LOSSY_NO_ALPHA ||
                this == WEBP_LOSSLESS_NO_ALPHA;
    }
}

File: app/src/main/java/com/example/doodle/remote/media/MediaType.kt
package com.example.doodle.remote.media

enum class MediaType(val mime: String, val extension: String) {
    UNKNOWN("", ""),
    MP4("video/mp4", "mp4"),
    MOV("video/quicktime", "mov"),
    _3GP("video/3gpp", "3gp"),
    MKV("video/x-matroska", "mkv"),
    GIF("image/gif", "gif"),
    JPG("image/jpeg", "jpg"),
    PNG("image/png", "png"),
    WEBP("image/webp", "webp"),
    BMP("image/bmp", "bmp"),
    HEIC("image/heic", "heic"),
    HEIF("image/heif", "heif");
}